content/client-lib-development-guide/features.textile

```diff
@@ -267,7 +268,23 @@ h3(#rest-channel). Channel
+** @(RSL6b)@ If, for example, incompatible encryption details are provided or invalid Base64 is detected in the message payload, an error message will be sent to the logger, but the message will still be delivered with last successful decoding and the @encoding@ field. For example, if a message had a decoding of "utf-8/cipher+aes-128-cbc/base64", and the payload was successfully Base64 decoded but the payload could not be decrypted because the @CipherParam@ details were not configured, the message would be delivered with a binary payload and an @encoding@ with the value "utf-8/cipher+aes-128-cbc". Additional steps need to be taken if decoding failed on "vcdiff" encoding, see @(RSL6c)@ 
+** @(RSL6c)@ Given "vcdiff" encoded deltas are applied to the previous message published on a channel, when a "vcdiff" encoding fails to be decoded, it makes it impossible for a client to apply subsequent deltas received from that point of failure forward. As such, the client must automatically execute the following recovery procedure in lieu of "RTL7e":#"RTL7e":
+*** @(RSL6c1)@ Log error with code 40018
+*** @(RSL6c2)@ Discard the message
+*** @(RSL6c3)@ Send an @ATTACH@ @ProtocolMessage@ with the channel serial set to the previous message to the message for which "vcdiff" decoding failed to the server, transitioning the channel state to @ATTACHING@, and waiting for a confirmation @ATTACHED@, as per @RTL4c@ and @RTL4f@
+*** @(RSL6c4)@ Any subsequent "vcdiff" decoding failures as per "RSL6c":#RSL6c while the recovery procedure from a previous failure as per "RSL6c3":#RSL6c3 is still in progress, i.e. the channel is in @ATTACHING@ state, should not trigger a new recovery procedure as per "RSL6c3":#RSL6c3.
+** @(RSL6d)@ The data payload of the last message on each channel, i.e. @Message.data@ has to be stored at all times since it might be needed to decode the "vcdiff" encoding. The last message bookkeeping rules are:
+*** @(RSL6d1)@ If the message encoding string ends in @base64@, the message should be base64-decoded (and the encoding string modified accordingly per "RSL6":#RSL6) before storing the last payload.
+*** @(RSL6d2)@ If "vcdiff" encoding is present the direct output of decoding the "vcdiff" encoding is stored as a last message, no matter that more encodings might be present
+** @(RSL6e)@ The @id@ of the last received message on each channel has to be stored. In case the next message received is delta message (i.e. contains @vcdiff@ encoding) the stored last message @id@ has to be compared against the @Message.extras.delta.from@ of the delta message. If these are different the delta application and hence the message decoding fail. The recovery procedure from "RSL6c":#RSL6c must be executed. 
```

Comment:
"is delta message" reads better as "is a delta message".
"hence the message decoding fail" needs a "should" before the word "fail".